src/cumulus/internal.clj

```diff
@@ -0,0 +1,302 @@
+(ns cumulus.internal
+  (:require
+    [clojure.string :as str]))
+
+
+(defn reqd
+  "Get required parameter k from map m. Throw exception if not found."
+  [m k]
+  (if (contains? m k)
+    (name (get m k))
```

Comment:
`name` cannot be applied to non (string | keyword | symbol), hence `name` should not be used here. Rather `name` should be applied at the call site where the values are anticipated to be keyword/symbol/string.